src/TLuaInterpreter.cpp

```diff
@@ -10754,6 +10755,55 @@ bool TLuaInterpreter::compileAndExecuteScript(const QString& code)
     }
 }
 
+// reformats given Lua code. In case of any issues, returns the original code as-is
+// issues could be invalid Lua code or the formatter code bugging out
+QString TLuaInterpreter::formatLuaCode(const QString &code)
+{
+    if (code.isEmpty()) {
+        return code;
+    }
+    lua_State* L = pIndenterState;
+    if (!L) {
+        qDebug() << "LUA CRITICAL ERROR: no suitable Lua execution unit found.";
+        return code;
+    }
+    
+    if (!validLuaCode(code)) {
+        return code;
+    }
+
+    QString escapedCode = code;
+    // escape backslashes so we can pass \n to the function
+    escapedCode.replace(QLatin1String("\\"), QLatin1String("\\\\"), Qt::CaseInsensitive);
+    // escape quotes since we'll be using quotes to pass data to the function
+    escapedCode.replace(QLatin1String("\""), QLatin1String("\\\""), Qt::CaseInsensitive);
+    // escape newlines so they don't interpreted as newlines, but instead get passed onto the function
+    escapedCode.replace(QLatin1String("\n"), QLatin1String("\\n"), Qt::CaseInsensitive);
+
+    QString thing = QString(R"(return get_formatted_code(get_ast("%1"), {indent_chunk = '  ', right_margin = 100, max_text_width = 160, keep_comments = true}))").arg(escapedCode);
+    int error = luaL_dostring(L, thing.toUtf8().constData());
+    QString n;
+    if (error != 0) {
+        string e = "no error message available from Lua";
+        if (lua_isstring(L, 1)) {
+            e = "Lua error:";
+            e += lua_tostring(L, 1);
+        }
+        if (mudlet::debugMode) {
+            qDebug() << "LUA ERROR: code did not compile: ERROR:" << e.c_str();
+        }
+        QString objectName = "error in Lua code";
+        QString functionName = "no debug data available";
+        logError(e, objectName, functionName);
```

Comment:
That `logError` seems to have an approximate signature of `(std::string, QString, QString)` strikes me as a little odd - I'd have expected to see all of one or the other _string_ types - and if the latter _are_ `QString`s  then why are we not initialising them with `QLatin1String(`...`)` (or `QStringLiteral(`...`)`)?